src/dg/dg_base.cpp

```diff
@@ -551,51 +557,56 @@ void DGBase<dim,real,MeshType>::assemble_cell_residual (
         auto current_face = current_cell->face(iface);
 
         // CASE 1: FACE AT BOUNDARY
```

Comment:
The logic for face residual assembly has become quite complicated. I mostly followed the existing structure, but I admit that I've contributed to some messy code!

I'm happy to take functionality suggestions, but I may just open an issue to refactor this section of code. It might be a good task to introduce a new student to the code.